### PR TITLE
Colossus/minor performance enhancements

### PR DESCRIPTION
**Pull request by mnaamani**
This PR primarily reduce overhead on IPFS node while syncing content. When a node first starts syncing content it is unreasonable to try to pin all content at one time. The overhead primarily on RAM is too much.
I applied a basic solution of limiting concurrent new pins to a fixed amount and shortening the sync run interval from 5min to 2min.

This code is running on the staging node `alexandria-testing-3.joystream.app`

### CODE REVIEW

File: storage-node/packages/colossus/lib/sync.js [hunk 1]
```diff
@@ -19,73 +19,100 @@
...
+
+          if (!synced && !syncing && contentBeingSynced.size < MAX_CONCURRENT_SYNC_ITEMS) {
+            try {
+              contentBeingSynced.set(contentId, true)
+
+              await storage.synchronize(contentId, () => {
+                contentBeingSynced.delete(contentId)
+              })
+            } catch (err) {
+              contentBeingSynced.delete(contentId)
```
Comment (shamil-gadelshin) [inline storage-node/packages/colossus/lib/sync.js]: Shouldn't we log an error message here?

File: storage-node/packages/colossus/lib/sync.js [hunk 2]
```diff
@@ -19,73 +19,100 @@
...
+            }
+          }
+        } catch (err) {
+          //
```
Comment (shamil-gadelshin) [inline storage-node/packages/colossus/lib/sync.js]: Shouldn't we log an error message here?

File: storage-node/packages/colossus/lib/sync.js [hunk 3]
```diff
@@ -19,73 +19,100 @@
...
+        // make sure the data object was Accepted by the liaison,
+        // don't just blindly attempt to fetch them
+
+        try {
+          const { synced, syncing } = await storage.syncStatus(contentId)
+
+          if (synced) {
+            return contentId
+          }
+
```
Comment (shamil-gadelshin) [inline storage-node/packages/colossus/lib/sync.js]: Do we need to check 'synced' again here?

File: storage-node/packages/util/ranges.js [hunk 4]
```diff
@@ -21,18 +21,22 @@
 const uuid = require('uuid')
 const streamBuf = require('stream-buffers')
```
Comment (shamil-gadelshin) [inline storage-node/packages/util/ranges.js:23]: Please, remove or uncomment all '// debug(' lines.

**Review by shamil-gadelshin — CHANGES_REQUESTED** [review]

File: storage-node/packages/colossus/lib/sync.js (hunk 1 above)
Comment (mnaamani) [reply storage-node/packages/colossus/lib/sync.js]: fixed in 
0a6d836

File: storage-node/packages/colossus/lib/sync.js (hunk 2 above)
Comment (mnaamani) [reply storage-node/packages/colossus/lib/sync.js]: fixed in 
0a6d836

File: storage-node/packages/colossus/lib/sync.js (hunk 3 above)
Comment (mnaamani) [reply storage-node/packages/colossus/lib/sync.js]: Improved the if/else blocks in
0a6d836

File: storage-node/packages/util/ranges.js (hunk 4 above)
Comment (mnaamani) [reply storage-node/packages/util/ranges.js:23]: done in 1b4d6d6

**Review by shamil-gadelshin — APPROVED** [review]
LGTM

**Comment by mnaamani** [comment]
Found a couple issues with these changes. Will push an update

**Comment by mnaamani** [comment]
There was a mistake in storage.synchronize() not always making the callback, making correct selection of new items to sync and. Fixed in https://github.com/Joystream/joystream/pull/1312/commits/6ec9463eadca63f1916a7031e8a667a2deb43eaf

**Review by shamil-gadelshin — APPROVED** [review]
LGTM

**Review by bwhm — APPROVED** [review]

**Pull request merged**